jaeger-core/src/test/java/com/uber/jaeger/samplers/RemoteControlledSamplerTest.java

```diff
@@ -108,9 +108,8 @@ public void testUpdatePerOperationSamplerUpdatesExistingPerOperationSampler() th
     undertest = new RemoteControlledSampler(SERVICE_NAME, samplingManager, perOperationSampler, metrics);
 
     undertest.updateSampler();
-    Thread.sleep(20);
     //updateSampler is hit once automatically because of the pollTimer
-    verify(perOperationSampler, times(2)).update(parameters);
+    verify(perOperationSampler, timeout(2000).times(2)).update(parameters);
```

Comment:
ELI5, how does this fix the flakiness?

Reply:
I assumed that the `Thread.sleep` was being set to a value too low was the cause of the test failure and changed it to use mockito's timeout, which would poll until timeout to see if the expectation was met. 

That being said, I haven't yet verified that my assumption is correct, and simply wanted to trigger a build on travis

Reply:
if it works similar to https://github.com/awaitility/awaitility, then it's a sensible approach to deal with flaky tests - the alternative is to build complicated locking/semaphores, which is easy to get wrong. But given that we're simply protecting against delays in thread scheduling on VMs, and that the mean delay is negligible, it makes sense to use longer timeouts, like 10sec (under normal circumstances it will finish a lot sooner).

However, if the flakiness of the test is not due to scheduling but fundamentally due to a race condition in the test, then the timeout doesn't help.